.evergreen/download-mongodb.sh

```diff
@@ -73,7 +73,7 @@ get_mongodb_download_url_for ()
          MONGODB_LATEST="https://fastdl.mongodb.org/sunos5/mongodb-sunos5-x86_64-latest.tgz"
              MONGODB_36=""
              MONGODB_34="https://fastdl.mongodb.org/sunos5/mongodb-sunos5-x86_64-${VERSION_34}.tgz"
-             MONGODB_32="https://fastdl.mongodb.org/sunos5/mongodb-sunos5-x86_64-${VERSION_32}.tgz"
```

Comment:
Let's hardcode this to the last available version to avoid breaking anyone. My tests show that to be https://fastdl.mongodb.org/sunos5/mongodb-sunos5-x86_64-3.2.14.tgz

Do the same for 3.4 and 3.0.